google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/ConnectionWorkerPool.java

```diff
@@ -65,6 +68,11 @@ public class ConnectionWorkerPool {
   private final Set<ConnectionWorker> connectionWorkerPool =
       Collections.synchronizedSet(new HashSet<>());
 
+  /*
+   * Contains the mapping from stream name to updated schema.
+   */
+  private Map<String, TableSchema> streamNameToUpdatedSchema = new ConcurrentHashMap<>();
```

Comment:
Is there a way to only cache to the level of table, the size of this map could be huge, if it is per stream one table schema.

Reply:
Changed to cache table name